app/clients/document_download.py

```diff
@@ -45,11 +40,13 @@ def upload_document(self, service_id, file_contents):
 
             response.raise_for_status()
         except requests.RequestException as e:
-            error = DocumentDownloadError.from_exception(e)
-            current_app.logger.warning(
-                'Document download request failed with error: {}'.format(error.message)
-            )
-
-            raise error
+            if e.response.status_code == 400:
```

Comment:
Optional: I feel that this would be a good place for an inline code comments. It's obviously very easy to spot that we have different behaviour for a 400, but it's very unclear WHY the behaviour is different for 400. Obviously people could go search back in commit history but not ideal